test/AssetManager.js

```diff
@@ -0,0 +1,167 @@
+      const j4 = await assetManager.getJob(4);
+      assertBNEqual(j1.result, toBigNumber('111'));
+      assertBNEqual(j2.result, toBigNumber('222'));
+      assertBNEqual(c3.result, toBigNumber('333'));
+      assertBNEqual(j4.result, toBigNumber('444'));
+    });
+
+    it('should not create a collection if one of the jobIDs is not a job', async function () {
+      const collectionName = 'Test Collection2';
+      const tx = assetManager.createCollection(collectionName, [1, 2, 5], 2);
+      await assertRevert(tx, 'Job ID not present');
+    });
+
+    it('should not create collection if it does not have more than 1 or any jobIDs', async function () {
+      const collectionName = 'Test Collection2';
+      const tx1 = assetManager.createCollection(collectionName, [], 1);
+      await assertRevert(tx1, 'Number of jobIDs low to create collection');
+      const tx2 = assetManager.createCollection(collectionName, [1], 1);
+      await assertRevert(tx2, 'Number of jobIDs low to create collection');
+    });
+
+    it('aggregationMethod should not be equal to 0 or greater than 3', async function () {
+      const url = 'http://testurl.com/4';
+      const selector = 'selector/4';
+      const name = 'test4';
+      const repeat = true;
+      await assetManager.createJob(url, selector, name, repeat);
+      const collectionName = 'Test Collection2';
+      const tx1 = assetManager.createCollection(collectionName, [1, 2, 5], 4);
+      await assertRevert(tx1, 'Aggregation range out of bounds');
+      const tx2 = assetManager.createCollection(collectionName, [1, 2, 5], 0);
+      await assertRevert(tx2, 'Aggregation range out of bounds');
+      await assetManager.createCollection(collectionName, [1, 2, 5], 1);
+    });
+
+    it('should not create collection if duplicates jobIDs are present', async function () {
+      const collectionName = 'Test Collection2';
+      const tx = assetManager.createCollection(collectionName, [1, 2, 2, 5], 1);
+      await assertRevert(tx, 'Duplicate JobIDs sent');
+    });
+
+    it('should not add jobID to a collection if the collectionID specified is not a collection', async function () {
+      const tx = assetManager.addJobToCollection(5, 4);
+      await assertRevert(tx, 'Collection ID not present');
+    });
+
+    it('should not add jobID to a collection if the jobID specified is not a Job', async function () {
+    // jobID does not exist
+      const tx = assetManager.addJobToCollection(3, 7);
+      await assertRevert(tx, 'Job ID not present');
+      // jobID specified is a collection
+      const tx1 = assetManager.addJobToCollection(3, 6);
+      await assertRevert(tx1, 'Job ID not present');
+    });
+
+    it('should not be add job if it already exists in the collection', async function () {
+      const tx = assetManager.addJobToCollection(3, 1);
+      await assertRevert(tx, 'Job exists in this collection');
+    });
+
+    // it('should be able to get result using proxy', async function () {
```

Comment:
Any reason to comment this @SamAg19 ?

Reply:
Yes, this was intentional as test cases on the delegator needs to be work upon and with these new changes that are bring introduced, this test case becomes redundant for now.